### PR TITLE
upgraded Python to 3.7

### PR DESCRIPTION
**Pull request by dmalan**

### CODE REVIEW

File: etc/profile.d/baseimage.sh [hunk 1]
```diff
@@ -7,3 +7,10 @@ fi
 alias pip="pip3 --no-cache-dir"
 alias pip3="pip3 --no-cache-dir"
 alias python="python3"
+
+# Flask
+export FLASK_APP="application.py"
+
+# PATH
+export PYENV_ROOT=/opt/pyenv
+export PATH=/opt/cs50/bin:/opt/bin:/usr/local/sbin:/usr/local/bin:"$PYENV_ROOT"/shims:"$PYENV_ROOT"/bin:/usr/sbin:/usr/bin:/sbin:/bin
```
Comment (kzidane) [inline etc/profile.d/baseimage.sh:16]: Wouldn't quoting the whole value be sufficient as opposed to quoting each `PYENV_ROOT` separately?

File: Dockerfile [hunk 2]
```diff
@@ -39,19 +39,49 @@ RUN curl --silent https://packagecloud.io/install/repositories/cs50/repo/script.
...
+        llvm \
+        make \
+        tk-dev \
+        wget \
+        xz-utils \
+        zlib1g-dev && \
+    wget -P /tmp https://github.com/pyenv/pyenv/archive/master.zip && \
+    unzip -d /tmp /tmp/master.zip && \
+    rm -f /tmp/master.zip && \
+    mv /tmp/pyenv-master "$PYENV_ROOT" && \
```
Comment (kzidane) [inline Dockerfile:76]: Any reason not to `git clone` into `PYENV_ROOT` directly?

File: etc/profile.d/baseimage.sh (hunk 1 above)
Comment (dmalan) [reply etc/profile.d/baseimage.sh:16]: Yup but feels sloppy, rather like putting more lines than necessary in a `try` block. Most folks seem to just quote vars themselves. Cf. http://tldp.org/LDP/abs/html/quotingvar.html.

> When referencing a variable, it is generally advisable to enclose its name in double quotes.

File: Dockerfile (hunk 2 above)
Comment (dmalan) [reply Dockerfile:76]: Didn't want the `.git` to be there (though we could just `rm`). And just tend to do this out of habit, ensures that the `mv` is atomic, so the dir's either there or not, not partially populated. Doesn't matter in a `Dockerfile` so much but in a script, feels best to move things into place only when ready.

File: etc/profile.d/baseimage.sh (hunk 1 above)
Comment (kzidane) [reply etc/profile.d/baseimage.sh:16]: I think this sentence might be referring to using variables in other contexts (e.g., passing them as command-line args, in a loop header, etc). Quoting individual variables in a *single* value that's assigned to a variable seems redundant to me tbh but no big deal.

**Review by kzidane — APPROVED** [review]

**Pull request merged**